### ISSUE TITLE
best learning rate

### ISSUE THREAD

**Issue by destlaver** [victorchall/EveryDream2trainer#105]
so I made some experiments changing the lr and I found out something strange.
The trainings seems to improve when the lr is as low as 1e-7
The loss decreases steadly and the sample images are more consistent.
Is it normal? Am I doing something wrong?

setup:
gpu rtx 4090 48vcpu, 124 gb ram
batch size: 10
lr_scheduler: cosine
lr: 1.2e-7
dataset size: 5k
captions type: tags

Also I'm about to train with an A100, 50k training set, and I'm not sure about the batch size and how can i finetune hyperparameters to take advantage of the entire gpu.

**Comment by victorchall** [comment]
SD1.5 seems to respond fine to ~1e-6, but SD2.1 based models may need far less.  There are lots of discussions on this on the Discord, usually a better place to connect with others.

**Comment by destlaver** [comment]
thank you sir. 
So I've started training on a 20k dataset with a A100.
I'm using SD1.4, a cosine scheduler and a lr of 1.2e-6. I've used cosine because I noticed the lower the lr the best the samples but I don't want too be far from 1e-6 nonetheless.
I think the training is going well but there are few things I don't understand: 

1- is it normal to get 1.41 it/s with an A100 while with an rtxA6000 I get like 1.12 it/s. Isn't it strange this little improvement
2 - I tried with double gpu but strangely it seems EveryDream only uses one
3 - I'm getting decent results with the bodies but the model still can't produce a not smashed face. How can I improve face generation?
4 - I left batch size untouched (10) 'cause I read online large batch size decrease training quality
5 - I'm doing training session of about 10 epochs, then I restart with the same hyperparameters and with the same dataset + additional new images

this are the performances:
[](https://ibb.co/HBHZRTY)

What do you think? I'm still a newbie and I would appreciate an advice. Thank you

**Comment by official-elinas** [comment]
#1 - An A100 is on par with an RTX 3070 in terms of raw performance, while an A6000 is closer to a 3090 Ti. I'm assuming you increased your batch size on the A100 so that seems right to me? A6000 10,752 vs A100 6,912 shading units.
#2 - Can't comment on this as I haven't used multi-gpu for everydream.
#3 - Use close up face shots for characters as you want variation in your model dataset.
#4 - You generally want to scale batch size with learning rate linearly. So, if you're doing a batch size of 10, your effective LR 1/10th of that which is fairly low.  https://www.baeldung.com/cs/learning-rate-batch-size

Hope that helps.

**Comment by victorchall** [comment]
1. it/s isn't total speed.  You need to multiply by batch size.  images per second is logged in tensorboard and the trainer calculates this for you.  It is also in the console log on the EPOCH line as "img/s".  That's a more meaningful measure to compare.  It's shown in your screenshots.  Higher resolution will reduces img/s.  SD2.1 is a bit slower as well compared to SD1.5 since it is a slightly larger model.  Turning on gradient checkpointing reduces img/s, but may let you run a larger batch size which can improve speed again.  You can toy with it to keep your batch size in the 6-10 range depending on resolution and VRAM.  On my 3090 I only use gradient checkpointing to run 768+ resolution so I can get the batch size back up to ~6-8.  You won't need gradient checkpointing on a 40GB or 48GB GPU.

2. No current support for multigpu, but I have a second 3090 now so I may be able to add this.  Maybe a while, give me maybe a month.  I need a new PSU as well to run it...

3.  Yes, make sure you have a good mix of high resolution face close ups.  
4.  A100 can run much larger batch size, reduces it/s but may get slightly more images per second.  Makes epochs fewer steps.  Running a really large batch size may change training behavior.  Usually suggested you bump LR up slightly with higher batch size.  For instance, if you normally run LR 1e-6 and that works for you on a 3090 at batch size around 5-10, you might be able to run the same settings but batch size 20-25 on an A100 but you may want to use about 1.5e-6. Many people report for specific subjects a big batch size trains worse, but it may be ok if you're training a really large data set, like 10s of thousands for a general style. 
5.  I would just include all data from the start.  If you have problems with face close ups you can run just the face close ups just a couple extra epochs only after training everything.

**Comment by victorchall** [comment]
I would recommend joining discord, there's a lot of suggestions and discussions there with an active community.  Very few people read/comment on github issues.

**Comment by destlaver** [comment]
thank you to both. I'm closing the issue.
